auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -183,29 +159,12 @@ public Page<ProductBidDetailsResponse> getProductByUserAndStatus(final UUID user
                 .map(ProductBidDetailsResponse::new);
     }
 
-    private PaymentInfoEntity handlePaymentInfo(ProductAddRequest productRequest) {
-        if (!productRequest.isDataChanged() && productRequest.getUserId() != null) {
-            final UserEntity user = userRepository.findById(productRequest.getUserId())
-                    .orElseThrow(() -> new ResourceNotFoundException("User with the given ID does not exist"));
-
-            return user.getPaymentInfo();
-        } else {
-            return createAndSavePaymentInfo(productRequest);
-        }
-    }
-
-    private PaymentInfoEntity createAndSavePaymentInfo(ProductAddRequest productRequest) {
-        PaymentInfoEntity paymentInfo = new PaymentInfoEntity();
-
-        paymentInfo.setAddress(productRequest.getAddress());
-        paymentInfo.setCity(productRequest.getCity());
-        paymentInfo.setZipCode(productRequest.getZipCode());
-        paymentInfo.setExpirationDate(productRequest.getExpirationDate());
-        paymentInfo.setCardNumber(productRequest.getCardNumber());
-        paymentInfo.setNameOnCard(productRequest.getNameOnCard());
-        paymentInfo.setCountry(productRequest.getCountry());
-
-        return paymentInfoRepository.save(paymentInfo);
+    // used to deactivate products that have expired using scheduled task
+    @Transactional
+    @Scheduled(cron = "0 0 0 * * *", zone = "UTC")
+    public void deactivateExpiredProducts() {
```

Comment:
It would be nice to have special util class for this
Where scheduled activities would be handled